cpp/src/column/column.cu

```diff
@@ -176,9 +177,77 @@ struct CreateColumnFromView {
 
 };
 
+struct CreateColumnFromViewVector {
+  std::vector<cudf::column_view> views;
+  cudaStream_t stream;
+  rmm::mr::device_memory_resource *mr;
+
+ template <typename ColumnType,
+           std::enable_if_t<std::is_same<ColumnType, cudf::string_view>::value>* = nullptr>
+ std::unique_ptr<column> operator()() {
+   std::vector<cudf::strings_column_view> sviews;
+   for (auto &v : views) { sviews.emplace_back(v); }
+
+   auto col = cudf::strings::detail::concatenate(sviews, mr, stream);
+
+   cudf::detail::concatenate_masks(views,
+       (col->mutable_view()).null_mask(), stream);
+
+   return std::move(col);
+ }
+
+ template <typename ColumnType,
+           std::enable_if_t<cudf::is_fixed_width<ColumnType>()>* = nullptr>
+ std::unique_ptr<column> operator()() {
+
+   auto type = views.front().type();
+   size_type total_element_count = 0;
+   for (auto &v : views) { total_element_count += v.size(); }
```

Comment:
This does not work because std::accumulate accepts a binary predicate lambda. This implies that it only works on vectors of the type of accumulation, in this case size_type.

Reply:
Example code :
```
class A {
  int _a;
  public:
    A(int a) : _a(a) {}
  int val() { return _a; }
};

int main(int argc, char *argv[]) {
  std::vector<int> a(10, 0);
  std::vector<int> b(5, 7);
  std::vector<int> c(5, 3);
  std::vector<std::vector<int>> V{a, b, c};
  //--Commented out code fails to compile
  //size_t l =
  //std::accumulate(V.begin(), V.end(), 0,
  //[](auto &v) { return v.val(); });
  return 0;
}

```